src/script/engine.zig

```diff
@@ -624,6 +624,33 @@ test "Script execution - OP_1 OP_2 OP_IFDUP" {
     try std.testing.expectEqualSlices(u8, &[_]u8{2}, element1);
 }
 
+test "Script execution - OP_OVER" {
+    const allocator = std.testing.allocator;
+
+    // Simple script: OP_1 OP_2 OP_3 OP_OVER
+    const script_bytes = [_]u8{ 0x51, 0x52, 0x53, 0x78 }; // 0x78 is OP_OVER
+    const script = Script.init(&script_bytes);
+
+    var engine = Engine.init(allocator, script, .{});
+    defer engine.deinit();
+
+    try engine.execute();
+
+    // Ensure the stack has the expected number of elements
+    try std.testing.expectEqual(@as(usize, 4), engine.stack.len());
+
+    // Check the stack elements
+    const element0 = try engine.stack.peek(0);
+    const element1 = try engine.stack.peek(1);
+    const element2 = try engine.stack.peek(2);
+    const element3 = try engine.stack.peek(3);
```

Comment:
```suggestion
    const element0 = try engine.stack.peekInt(0);
    const element1 = try engine.stack.peekInt(1);
    const element2 = try engine.stack.peekInt(2);
    const element3 = try engine.stack.peekInt(3);
```

And then `2` instead of `&[_]u8{2}` and expectEqual rather than expectEqualSlices